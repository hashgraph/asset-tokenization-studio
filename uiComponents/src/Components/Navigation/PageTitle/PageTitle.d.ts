/// <reference types="react" />
import type { BaseMultiStyleConfiguration } from "@Theme/Components/BaseMultiStyleConfiguration";
import type { FlexProps } from "@chakra-ui/react";
import type { TextProps } from "@/Components/Foundations/Text";
import type { IconButtonProps, IconButtonThemeConfiguration } from "@Components/Interaction/IconButton";
export declare const pageTitlePartsList: Array<"backButton" | "textTitle" | "container">;
type Parts = typeof pageTitlePartsList;
export interface PageTitleThemeConfiguration extends BaseMultiStyleConfiguration<Parts> {
    backButton?: IconButtonThemeConfiguration;
}
export interface PageTitleProps extends Omit<FlexProps, "children"> {
    title?: string;
    onGoBack?: () => void;
    backButtonProps?: Omit<IconButtonProps, "icon" | "variant">;
    titleProps?: TextProps;
    isLoading?: boolean;
}
export declare const PageTitle: ({ title, onGoBack, backButtonProps, titleProps, isLoading, }: PageTitleProps) => JSX.Element;
export {};
