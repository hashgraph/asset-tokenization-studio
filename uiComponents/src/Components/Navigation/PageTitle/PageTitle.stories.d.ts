import type { PageTitleProps } from "@Components/Navigation/PageTitle";
declare const _default: import("@storybook/types").ComponentAnnotations<import("@storybook/react/dist/types-0a347bb9").R, PageTitleProps>;
export default _default;
export declare const WithBackButton: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, PageTitleProps>;
export declare const WithoutBackButton: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, PageTitleProps>;
export declare const WithLoading: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, PageTitleProps>;
