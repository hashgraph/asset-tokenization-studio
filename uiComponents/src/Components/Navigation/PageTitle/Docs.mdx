import {
  Story,
  Meta,
  Title,
  ArgTypes,
  Primary,
  Stories,
  Source,
} from "@storybook/addon-docs";
import { Grid, GridItem, Text, Link, Center, Flex } from "@chakra-ui/react";
import * as PageTitleStories from "./PageTitle.stories";

<Meta of={PageTitleStories} />

<Title />

<a
  href="https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/NEW---ioBricks-Library?node-id=2091-20098&t=7YJ1vWLNVVuMBa0t-0"
  target="\_blank"
>
  View figma design
</a>

<Primary />

## How to set up

### Config theme

Add the following configurations to the `theme.tsx` and `PageTitle.ts` files, respectively.

```tsx dark
const theme = {
  components: {
    PageTitle: ConfigPageTitle,
  },
};
```

```tsx dark
const ConfigPageTitle: PageTitleThemeConfiguration = {
  parts: pageTitlePartsList,
  baseStyle: {},
};
```

### Component parts

You have these parts available: `"backButton" | "textTitle" | "container"` (`pageTitlePartsList`)

## Props

<ArgTypes of={PageTitleStories} />

## Stories

<Stories title="" />
