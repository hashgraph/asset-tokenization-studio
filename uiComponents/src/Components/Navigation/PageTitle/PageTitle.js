var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from "react";
import { Flex, SkeletonText, useMultiStyleConfig as useChakraMultiStyleConfig, } from "@chakra-ui/react";
import { ArrowLeft } from "@phosphor-icons/react";
import { PhosphorIcon } from "@/Components/Foundations/PhosphorIcon";
import { Text } from "@/Components/Foundations/Text";
import { IconButton } from "@Components/Interaction/IconButton";
export var pageTitlePartsList = ["backButton", "textTitle", "container"];
export var PageTitle = function (_a) {
    var title = _a.title, onGoBack = _a.onGoBack, backButtonProps = _a.backButtonProps, titleProps = _a.titleProps, isLoading = _a.isLoading;
    var styles = useChakraMultiStyleConfig("PageTitle");
    return (React.createElement(Flex, { sx: styles.container },
        onGoBack && (React.createElement(IconButton, __assign({ sx: styles.backButton, icon: React.createElement(PhosphorIcon, { as: ArrowLeft, size: "xxs" }), "aria-label": "Go back", onClick: onGoBack, variant: "secondary", size: "md" }, backButtonProps))),
        isLoading ? (React.createElement(SkeletonText, { skeletonHeight: 6, noOfLines: 1, minWidth: 25 })) : (React.createElement(Text, __assign({ sx: styles.textTitle }, titleProps), title))));
};
