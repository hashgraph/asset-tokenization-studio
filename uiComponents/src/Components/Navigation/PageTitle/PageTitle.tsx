import React from "react";
import type { BaseMultiStyleConfiguration } from "@Theme/Components/BaseMultiStyleConfiguration";
import type { FlexProps } from "@chakra-ui/react";
import {
  Flex,
  SkeletonText,
  useMultiStyleConfig as useChakraMultiStyleConfig,
} from "@chakra-ui/react";
import { ArrowLeft } from "@phosphor-icons/react";
import type { SystemStyleObject as ChakraSystemStyleObject } from "@chakra-ui/system";
import { PhosphorIcon } from "@/Components/Foundations/PhosphorIcon";
import type { TextProps } from "@/Components/Foundations/Text";
import { Text } from "@/Components/Foundations/Text";
import type {
  IconButtonProps,
  IconButtonThemeConfiguration,
} from "@Components/Interaction/IconButton";
import { IconButton } from "@Components/Interaction/IconButton";

export const pageTitlePartsList: Array<
  "backButton" | "textTitle" | "container"
> = ["backButton", "textTitle", "container"];

type Parts = typeof pageTitlePartsList;

export interface PageTitleThemeConfiguration
  extends BaseMultiStyleConfiguration<Parts> {
  backButton?: IconButtonThemeConfiguration;
}

type PageTitleThemeStyle = Record<Parts[number], ChakraSystemStyleObject>;

export interface PageTitleProps extends Omit<FlexProps, "children"> {
  title?: string;
  onGoBack?: () => void;
  backButtonProps?: Omit<IconButtonProps, "icon" | "variant">;
  titleProps?: TextProps;
  isLoading?: boolean;
}

export const PageTitle = ({
  title,
  onGoBack,
  backButtonProps,
  titleProps,
  isLoading,
}: PageTitleProps) => {
  const styles = useChakraMultiStyleConfig("PageTitle") as PageTitleThemeStyle;

  return (
    <Flex sx={styles.container}>
      {onGoBack && (
        <IconButton
          sx={styles.backButton}
          icon={<PhosphorIcon as={ArrowLeft} size="xxs" />}
          aria-label="Go back"
          onClick={onGoBack}
          variant="secondary"
          size="md"
          {...backButtonProps}
        />
      )}
      {isLoading ? (
        <SkeletonText skeletonHeight={6} noOfLines={1} minWidth={25} />
      ) : (
        <Text sx={styles.textTitle} {...titleProps}>
          {title}
        </Text>
      )}
    </Flex>
  );
};
