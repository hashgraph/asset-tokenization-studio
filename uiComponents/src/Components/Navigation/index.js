export * from "./Breadcrumb";
export * from "./Header";
export * from "./PageTitle";
export * from "./Sidebar";
