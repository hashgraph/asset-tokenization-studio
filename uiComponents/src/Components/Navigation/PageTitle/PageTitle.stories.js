var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from "react";
import { Title, Subtitle, Story } from "@storybook/addon-docs";
import { Link, Flex } from "@chakra-ui/react";
import { PageTitle } from "@Components/Navigation/PageTitle";
var props = {
    title: "Page Title",
    onGoBack: undefined
};
export default {
    title: "Design System/Navigation/PageTitle",
    component: PageTitle,
    args: {
        isLoading: false
    },
    parameters: {
        docs: {
            page: function () { return (React.createElement(React.Fragment, null,
                React.createElement(Title, null, "Molecules & Organisms: Navigation"),
                React.createElement(Subtitle, null, "Page Title with back button"),
                React.createElement(Flex, { margin: "50px 0" },
                    React.createElement(Story, { id: "design-system-navigation-pagetitle--with-back-button" })),
                React.createElement(Link, { href: "https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/NEW---ioBricks-Library?node-id=2091-20098&t=7YJ1vWLNVVuMBa0t-0", isExternal: true }, "View figma design"))); }
        }
    }
};
var Template = function (args) { return React.createElement(PageTitle, __assign({}, args)); };
export var WithBackButton = Template.bind({});
WithBackButton.args = __assign(__assign({}, props), { onGoBack: function () {
        alert("Go back");
    } });
export var WithoutBackButton = Template.bind({});
WithoutBackButton.args = __assign({}, props);
export var WithLoading = Template.bind({});
WithLoading.args = __assign(__assign({}, props), { onGoBack: function () {
        alert("Go back");
    }, isLoading: true });
