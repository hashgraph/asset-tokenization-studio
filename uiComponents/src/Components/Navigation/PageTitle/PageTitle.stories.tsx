import React from "react";
import { Title, Subtitle, Story } from "@storybook/addon-docs";
import { Link, Flex } from "@chakra-ui/react";

import type { Meta, StoryFn } from "@storybook/react";
import type { PageTitleProps } from "@Components/Navigation/PageTitle";
import { PageTitle } from "@Components/Navigation/PageTitle";

const props: PageTitleProps = {
  title: "Page Title",
  onGoBack: undefined,
};

export default {
  title: "Design System/Navigation/PageTitle",
  component: PageTitle,
  args: {
    isLoading: false,
  },
  parameters: {
    docs: {
      page: () => (
        <>
          <Title>Molecules & Organisms: Navigation</Title>
          <Subtitle>Page Title with back button</Subtitle>

          <Flex margin="50px 0">
            <Story id="design-system-navigation-pagetitle--with-back-button" />
          </Flex>

          <Link
            href="https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/NEW---ioBricks-Library?node-id=2091-20098&t=7YJ1vWLNVVuMBa0t-0"
            isExternal
          >
            View figma design
          </Link>
          {/* TODO: fix, now documentation should go in another file */}
          {/* <ArgsTable
            story={CURRENT_SELECTION}
            exclude={[
              "onGoBack",
              "titleProps",
              "_firstLetter",
              "backButtonProps",
            ]}
          /> */}
        </>
      ),
    },
  },
} as Meta<typeof PageTitle>;

const Template: StoryFn<typeof PageTitle> = (args) => <PageTitle {...args} />;

export const WithBackButton = Template.bind({});

WithBackButton.args = {
  ...props,
  onGoBack: () => {
    alert("Go back");
  },
};

export const WithoutBackButton = Template.bind({});

WithoutBackButton.args = {
  ...props,
};

export const WithLoading = Template.bind({});
WithLoading.args = {
  ...props,
  onGoBack: () => {
    alert("Go back");
  },
  isLoading: true,
};
