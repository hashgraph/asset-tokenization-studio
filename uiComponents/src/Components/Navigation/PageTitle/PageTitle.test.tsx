import { render } from "@/test-utils";
import { fireEvent } from "@testing-library/react";
import React from "react";
import type { PageTitleProps } from "@Components/Navigation/PageTitle";
import { PageTitle } from "@Components/Navigation/PageTitle";

describe("<PageTitle/>", () => {
  const defaultText = "Content of the button";
  const defaultProps: PageTitleProps = {
    title: defaultText,
  };

  const factoryComponent = (props?: Partial<PageTitleProps>) =>
    render(<PageTitle {...defaultProps} {...props} />);

  test("renders correctly", () => {
    const component = factoryComponent();

    expect(component.asFragment()).toMatchSnapshot();
    expect(component.getByText(defaultText)).toBeVisible();
  });

  test("Should not render IconButton ", () => {
    const component = factoryComponent({
      ...defaultProps,
    });

    expect(component.queryByRole("button")).not.toBeInTheDocument();
  });

  test("Should call onClickBack function ", () => {
    const props = {
      ...defaultProps,
      onGoBack: jest.fn(),
    };
    const component = factoryComponent({
      ...props,
    });

    fireEvent.click(component.getByRole("button"));

    expect(props.onGoBack).toBeCalledTimes(1);
  });
  test("Should render loading skeleton ", () => {
    const component = factoryComponent({
      isLoading: true,
    });

    expect(component.container).toMatchSnapshot("WithLoading");
  });
});
