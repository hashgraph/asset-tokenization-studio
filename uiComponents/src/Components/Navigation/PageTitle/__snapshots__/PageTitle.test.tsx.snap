// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageTitle/> Should render loading skeleton : WithLoading 1`] = `
<div>
  <div
    class="css-1dtnjt5"
  >
    <div
      class="chakra-skeleton__group css-1ksujjj"
    >
      <div
        class="chakra-skeleton css-1nd92f6"
      />
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;

exports[`<PageTitle/> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="css-1dtnjt5"
  >
    <p
      class="css-1ppjmwm"
    >
      Content of the button
    </p>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
